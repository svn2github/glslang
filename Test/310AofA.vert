#version 310 es

// Check name mangling of functions with parameters that are multi-dimensional arrays.

#define NX 2
#define NY 3
#define NZ 4
void f(bool a, float b, uint[4] c, int[NY][NX] d) {
}

void main() {
  int[NY][NX] d;
  f(false, 12.1, uint[NZ](uint(0),uint(1),uint(1),uint(2)), d);
}
