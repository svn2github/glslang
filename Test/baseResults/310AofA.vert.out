310AofA.vert
Warning, version 310 is not yet complete; most version-specific features are present, but some are missing.

Shader version: 310
0:? Sequence
0:8  Function Definition: f(b1;f1;u1[4];i1[3][2]; (global void)
0:8    Function Parameters: 
0:8      'a' (in bool)
0:8      'b' (in highp float)
0:8      'c' (in 4-element array of highp uint)
0:8      'd' (in 3-element array of 2-element array of highp int)
0:11  Function Definition: main( (global void)
0:11    Function Parameters: 
0:?     Sequence
0:13      Function Call: f(b1;f1;u1[4];i1[3][2]; (global void)
0:13        Constant:
0:13          false (const bool)
0:13        Constant:
0:13          12.100000
0:13        Constant:
0:13          0 (const uint)
0:13          1 (const uint)
0:13          1 (const uint)
0:13          2 (const uint)
0:13        'd' (temp 3-element array of 2-element array of highp int)
0:?   Linker Objects
0:?     'gl_VertexID' (gl_VertexId highp int VertexId)
0:?     'gl_InstanceID' (gl_InstanceId highp int InstanceId)


Linked vertex stage:


Shader version: 310
0:? Sequence
0:8  Function Definition: f(b1;f1;u1[4];i1[3][2]; (global void)
0:8    Function Parameters: 
0:8      'a' (in bool)
0:8      'b' (in highp float)
0:8      'c' (in 4-element array of highp uint)
0:8      'd' (in 3-element array of 2-element array of highp int)
0:11  Function Definition: main( (global void)
0:11    Function Parameters: 
0:?     Sequence
0:13      Function Call: f(b1;f1;u1[4];i1[3][2]; (global void)
0:13        Constant:
0:13          false (const bool)
0:13        Constant:
0:13          12.100000
0:13        Constant:
0:13          0 (const uint)
0:13          1 (const uint)
0:13          1 (const uint)
0:13          2 (const uint)
0:13        'd' (temp 3-element array of 2-element array of highp int)
0:?   Linker Objects
0:?     'gl_VertexID' (gl_VertexId highp int VertexId)
0:?     'gl_InstanceID' (gl_InstanceId highp int InstanceId)

